# Exam 3

Just generic python setup first, including some helper functions to help with degree-radians conversion, and solving quadratic equations

In [3]:
import numpy as np
from numpy import sin, cos, tan, arctan2, sqrt

In [4]:
def rad(th):
    return th * np.pi/180

def deg(th):
    return th * 180/np.pi

def quadratic(a, b, c):
    return ((-b + sqrt(b**2 - 4*a*c)) / (2*a),
            (-b - sqrt(b**2 - 4*a*c)) / (2*a))

## Problem 1

I can use conservation of mechanical energy, though I have to take into account the work done by non-conservative forces. Specifically, that means friction here -- there is also the normal force, but it does not do any work here since it's perpendicular to the actual motion.

$$
E_f = E_i + W_{nc}
$$

At the final point, the block has neither kinetic nor gravitational potential energy, $E_f = 0$. At the initial point, it has both: $E_i = \frac{1}{2}mv_i^2 + mgh_i$. $v_i$ is what I'm looking to find. The initial height I can find using trig: $h_i = L \sin(30^\circ)$.

Since the force of friction is directly opposite to the motion, I can find the work done by friction to be $W_f = -F_f \Delta x = -\mu F_N \Delta x$.

There are two sections where friction is doing negative work -- they differ in the normal force. On the inclined plane, I should draw a FBD and find $F_N = mg\cos(30^circ)$. On the flat section it's just $F_N = mg$.

Putting it all together:
$$
0 = \frac{1}{2}mv_i^2 + mgL\sin(30^\circ) - \mu mg\cos(30^\circ) L - \mu mg d
$$

The mass cancels out and I can solve for $v_i$:

$$v_i = \sqrt{-2gL\sin(30^\circ) + 2\mu g(d + L\cos(30^\circ)}
$$

In [10]:
L = 100
d = 500
mu = .1
g = 10
v_i = (-2*g*L*sin(rad(30)) + 2*mu*g*(d + L*cos(rad(30))))**.5
print(f"v_i = {v_i:.2f} m/s")

v_i = 13.16 m/s


# Problem 2

## Part a)

Momentum of the sytem of the two cars is (approximately) conserved since it is a collision, ie., it happens fast. The only external forces during that short collision time are gravity, normal force and friction, which effect negligibly little change in momentum in that time (in fact, normal force in the vertical direction cancels gravity, so those don't do anything even over longer timespans).

Energy is also conserved in the situation, since spring force is conservative and we can use spring potential energy. In fact, we don't really even have to use, since the spring is relaxed before and after the collision, so that potential energy is zero.

## Part b)

So we have
$$
m_1 v_{1i} + m_2 v_{2i} = m_1 v_{1f} + m_2 v_{2f}
$$
and
$$
\frac{1}{2} m_1 v_{1i}^2 + \frac{1}{2} m_2 v_{2i}^2 = \frac{1}{2} m_1 v_{1f}^2 + \frac{1}{2} m_2 v_{2f}^2
$$
We know that $v_{2i} = 0$, $v_{1i} = 3 m/s$, $m_1 = 1 kg$ and $m_2 = 2 kg$.

So we have two equations for two unknowns, which can be solved, but I'm not going to type it down again here. It is noteworthy, though, that because one equation is quadratic, there are two possible solutions. One in fact is for the final velocities to equal the respective initial velocities, which makes sense since when no collision happens, momentum and energy are also conserved. If one does the math as I did in class, I ended up dividing by $v_{1f} - v_{1i}$, which is how that solution "disappeared", since one can't divide by zero, which would be the case for the no-collision case. In any case, the solution we want is:

\begin{array}{ccc}
v_{1f}&=&{\dfrac {m_{1}-m_{2}}{m_{1}+m_{2}}}v_{1i}\\
v_{2f}&=&{\dfrac {2m_{1}}{m_{1}+m_{2}}}v_{1i}
\end{array}


In [13]:
v_1i = 3
m_1 = 1
m_2 = 2
v_1f = (m_1 - m_2)/(m_1 + m_2) * v_1i
v_2f = (2 * m_1)/(m_1 + m_2) * v_1i
print(f"v_1f = {v_1f:.2f} m/s  v_2f = {v_2f:.2f} m/s")

v_1f = -1.00 m/s  v_2f = 2.00 m/s


# Problem 3

## Part a)

Since the gravitional force give by Newton's Law of Gravity is not constant, we need to actually compute the integral here: At least it's 1-d, and it's worth realizing that if we take the radial direction to be positive outward, the force of gravity points inward toward the center of the star.

$$
W_G = - \int_{r_A}^{r_B} F_G dr = 
- C M \int_{r_A}^{r_B} \frac{1}{r^2} dr = \left .CM\frac{1}{r}\right|_{r_A}^{r_B} = CM\left(\frac{1}{r_B} - \frac{1}{r_A}\right)
$$

Plugging in numbers:

In [19]:
r_A = 20 * 1000
r_B = 8  * 1000
C = 5 * 1000**2
M = 1000
W_G = C*M*(1/r_B - 1/r_A)
print(f"W_G = {W_G:.2f} Nm")

W_G = 375000.00 Nm


## Part b)

Since now the motion is perpendicular to the force of gravity, it doesn't do any work between B and C.

## Part c)

No need to fire the engines -- since gravity doesn't do any work, the starship's kinetic energy is conserved, ie., it moves at constant speed.

To figure out the speed that that's happening at, a FBD wouldn't hurt, though there's only one force, and that provides centripetal acceleration:

$$
M\frac{v^2}{r_B} = M a_c = F_{net} = F_G = \frac{CM}{r_B^2}
$$

Mass cancels out and we find
$$
v = \sqrt{C/r_B}
$$


In [20]:
v = (C/r_B)**.5
print(f"v = {v:.2f} m/s")

v = 25.00 m/s


# Problem 4

## Part a)

This being a collision, momentum is approximately conserved for the system of the two cars (see also the answer to the elastic collision problem).
Energy is not conserved, since the cars end up deforming and sticking together.

## Part b)

Since the cars stick together, both have the same final velocity. Momentum is conserved:

$$
m_1 \vec v_{1i} + m_2 \vec v_{2i} = (m_1 + m_2) \vec v_f
$$

which is easily solved for $\vec v_f$:

$$
\vec v_f = \frac{m_1 \vec v_{1i} + m_2 \vec v_{2i}}{m_1 + m_2}
$$


In [34]:
m_1 = 1500
m_2 = 2000
v_1i = np.array([3, 6])
v_2i = np.array([2, -5])
v_f = (m_1 * v_1i + m_2 * v_2i) / (m_1 + m_2)
print(f"v_f in components = {v_f} m/s")
print(f"v_f magnitude = {np.linalg.norm(v_f):.2f} m/s")
print(f"v_f angle = {deg(np.arctan2(v_f[1], v_f[0])):.2f} deg (N of E)")

v_f in components = [ 2.42857143 -0.28571429] m/s
v_f magnitude = 2.45 m/s
v_f angle = -6.71 deg (N of E)


So it's really 6.71$^\circ$ S of E.

# Problem 5

## Part a)

Similarly to the ballistic pendulum: During the first time interval, momentum is conserved (it's a completely inelastic collision; energy is not conserved). During the 2nd interval, energy is conserved (spring force is conservative, frictionless), but momentum is not for the block with embedded bullet, since the spring force is external.

## Part b)

Calling just before the collison "A", just after "B", and when the block comes to rest "C", I have conservation of momentum von A to B:

$$
m v_{A} = (m + M) v_{B}
$$
where $m$ is the mass of the bullet, $M$ the mass of the block, and $v_A$ the initial velocity of the bullet (the block is at rest, no momentum), $v_B$ the velocity of block with embedded bullet after the collision.

So 
$$
v_{A} = \frac{(m + M)}{m} v_{B}
$$

From B to C, conservation of energy
$$
\frac{1}{2} (m + M) v_B^2 = \frac{1}{2}kd^2
$$

So
$$
v_B = \sqrt{\frac{k}{m+M}} d
$$

Putting it together:

$$
v_A = \sqrt{k(m+M)}\frac{d}{m}
$$

So we need to know the spring constant and the two masses (and the distance the block slides).